### ISSUE TITLE
In some cases orca reads spaces instead of reading tabs.

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#104673]
Issue Type: <b>Bug</b>

This was previously reported on the orca list and appears to be related to the upgrade to electron 9.
Steps:
1. Launch orca.
1. turn on Speak indentation and justification in the tab speech of orca's preferences page.
1. Open an empty file in VSCode.
1. Press the tab and type this is a test.
1. Press the enter key.
1. Browse the text using up and down arrow.

Orca reads the text preceded by the number of leading spaces in the line, as expected.

1. Convert the indentation to tabs.
1. Browse the text using up and down arrow.

Orca should read the text preceded by the number of initial tabs in the line but instead it announces that there is a space at the beginning of the line.

On the orca list @joanie commented the following:

>This commit fixes the problem -- though it's not a specific fix for
>it: https://chromium.googlesource.com/chromium/src/+/e49659287. Not
>sure if it should be cherry-picked into Electron 9 or not....

VS Code version: Code - Insiders 1.49.0-insider (bd08768fd310d34cdd4423a53e5f3ca77340fea5, 2020-08-14T06:41:56.571Z)
OS version: Linux x64 5.7.12-arch1-1

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz (4 x 2786)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: unavailable_off<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|0, 1, 1|
|Memory (System)|9.60GB (1.12GB free)|
|Process Argv|-n --no-sandbox --crash-reporter-id 8d5a3738-d241-4b34-b208-057882a7b877|
|Screen Reader|yes|
|VM|0%|
|DESKTOP_SESSION|gnome-xorg|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome-xorg|
|XDG_SESSION_TYPE|x11|
</details>
We have written the needed data into your clipboard because it was too large to send. Please paste.

**Comment by isidorn** [comment]
@jvesouza thanks a lot for filling this.
So it seems like this is fixed in later versions of Chrome. And does not seem super critical that we cherry pick.
Though still leaving up the decision to @deepak1556 if the cherry-pick is low risk

**Comment by lukaszgo1** [comment]
This can also be reproduced on Windows with NVDA using latest VS Code Insider version. I'm afrait I have to disagree with the statement from the commend above that it's not worth cherry picking a fix for this to the Electron version used by VSCode. When working on a code which uses tabs for indent and spaces  for alignment it is imperative to tell if I'm, dealing with a tab or with a space.

**Comment by isidorn** [comment]
@lukaszgo1 yes, we still plan to include this fix.
@deepak1556 let me know if I can help somehow

**Comment by deepak1556** [comment]
> This commit fixes the problem -- though it's not a specific fix for
it: https://chromium.googlesource.com/chromium/src/+/e49659287. Not
sure if it should be cherry-picked into Electron 9 or not....

The CL in question seems to be part of major refactor https://bugs.chromium.org/p/chromium/issues/detail?id=1049261 . I am hesitant to backport it without more confirmation on the stability of the CL, is it okay to just cherry-pick this commit ? The changes went in Chromium 85 and our current insiders is on 83, a CL of this magnitude usually involves a certain risk factor for backport, especially even more if its part of refactor. I lack the context on this accessibility codepath to begin backport.

**Comment by isidorn** [comment]
@deepak1556 thanks for providing details. So if I understand Electron numbering correctly, this would be fixed in Electron 11.
I think it is fine to wait a couple of milestones for this, especially since users on Win and Mac do not seem to be hitting this in VS Code - at least nobody complained.

**Comment by isidorn** [comment]
@deepak1556 thanks for getting my electron numbering right :)
Thus I have create a new label fixed-in-electron-10

**Comment by RDMurray** [comment]
I also have this issue on Windows using NVDA or Narrator.

**Comment by isidorn** [comment]
@deepak1556 this seems to be a bigger issue than I originally thought, since more users are hitting this. Do we have some timeline on Electron 10 update? If it is far out do we have more details if it is just ok to cherry pick that larger fix? Thanks a lot.

**Comment by deepak1556** [comment]
We plan to skip 10 and go for 11, and it won't hit stable anytime before November.

>  If it is far out do we have more details if it is just ok to cherry pick that larger fix? 

I can give it a try, mostly OOF this week, will check back on this next week.

**Comment by isidorn** [comment]
Thanks a lot

**Comment by isidorn** [comment]
@deepak1556 we should move this to October?

**Comment by pranavlal** [comment]
Hi all,
I am adding to the chorus of voices stating that this bug is prevalent on Windows too using NVDA. Is there any kind of workaround? I am on the insider build channel.
Is there any kind of temporary  fix? I would prefer not to switch editors because my workflow would be significantly hampered if I had to do that.

**Comment by isidorn** [comment]
@pranavlal unforutnelty there is no workaround now. We plan to fix this as part of the Electron update, I will also check with Deepak.

**Comment by leonardder** [comment]
Sorry for opening the duplicate.
To be honest, I'm experiencing this as a pretty major issue. There's no reliable way to find out whether I'm indenting with tab or space, unless I copy stuff to another text editor.

**Comment by isidorn** [comment]
Let's add important label to make it clear this is a big blocker for our screen reader users.

**Comment by isidorn** [comment]
We have decided to build an exploration build with a fix for this on the electron side as a workaround for this
Next milestone we plan to backport the fix to master so we have a full milestone to self host on the fix. Since the fix is not trivial. Thus moving this to November

@deepak1556 will provide a link to the exploration build once we have it

**Comment by deepak1556** [comment]
Exploration builds:

[Zip Archive](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-x64-1603350397.tar.gz)
[Deb](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration_1.51.0-1603349218_amd64.deb)
[Snap](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-1603350775.snap)
[RPM](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-1.51.0-1603349307.el7.x86_64.rpm)

**Comment by jvesouza** [comment]
> Exploration builds:
> 
> [Zip Archive](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-x64-1603350397.tar.gz)
> [Deb](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration_1.51.0-1603349218_amd64.deb)
> [Snap](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-1603350775.snap)
> [RPM](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-1.51.0-1603349307.el7.x86_64.rpm)


Please forgive my ignorance, but I have some questions regarding the exploration build.

* Is the generation done every day or every night?
* Does the generation incorporate the corrections made in the master branch?
* If I find any problem/bug can I report normally?

Thanks.

**Comment by alex19EP** [comment]
> 
> 
> > Exploration builds:
> > [Zip Archive](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-x64-1603350397.tar.gz)
> > [Deb](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration_1.51.0-1603349218_amd64.deb)
> > [Snap](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-1603350775.snap)
> > [RPM](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-1.51.0-1603349307.el7.x86_64.rpm)
> 
> Please forgive my ignorance, but I have some questions regarding the exploration build.
> 
>     * Is the generation done every day or every night?
> 

this version won't auto update. 

>     * Does the generation incorporate the corrections made in the master branch?
> 

no. exploration builds are slice of master. at the time of generation. wich
happens by hand.

>     * If I find any problem/bug can I report normally?
> 

i think yes. but be aware that exploration builds will not update and may
contain bugs witch already fixed in master.

**Comment by deepak1556** [comment]
> Is the generation done every day or every night?

Yes current exploration branch tracks master, there is a scheduled build every day.

> If I find any problem/bug can I report normally?

Yup you can follow the usual issue reporting flow.

**Comment by alex19EP** [comment]
@deepak1556 then I was wrong in my conclusions. Is there any detailed documentation about exploration builds?

**Comment by jvesouza** [comment]
I confirmm that the reported problem seems to be fixed in this build.

Also the bug #93056  seems to be fixed.

**Comment by deepak1556** [comment]
@alex19EP nope there isn't one.

@jvesouza thanks for confirming!

**Comment by jvesouza** [comment]
This also fix bug #108857

**Comment by jvesouza** [comment]
The version I'm using from build explorations seems to be causing some crash in orca.
After I start code-exploration orca stays mute and I need to restart orca.

**Comment by jvesouza** [comment]
Just to check if I'm not doing something wrong:
Today I downloaded the build exploration and according to the about dialog, the version was generated on October 22, commit 1dfc92ea9b990fad440fcf12ce46cb85897861ce.

Is there a newer version?

**Comment by isidorn** [comment]
I see a newer version from 7 hours ago that seems realesed. Though @deepak1556 would know better.

**Comment by deepak1556** [comment]
yeah the link points to a older version, you can just check for updates under the gear icon and it should update to the latest released a few hours back. I don't plan on updating the above link as the app will update anyways once you download it :)

**Comment by jvesouza** [comment]
@deepak1556 Thanks for the clarification, but I'm probably still doing something wrong.
When I try to update from code-exploration I am redirected to the following url:
https://code.visualstudio.com/insiders/

**Comment by isidorn** [comment]
Moving this to next milestone since then we plan to ship Electron 11 as part of Stable.
